### PR TITLE
Add Graph Interface

### PR DESCRIPTION
**Pull request by LHT129**
issue: #40

### CODE REVIEW

**Review by ShawnShawnYou — APPROVED** [review]

**Review by inabao — APPROVED** [review]

File: src/data_cell/graph_interface.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        this->max_capacity_ = std::max(this->max_capacity_, total_count_ + 1);
+        this->InsertNeighborsById(total_count_ + 1, neighbor_ids);
+        IncreaseTotalCount(1);
+        return total_count_;
+    }
+
+    virtual void
+    IncreaseTotalCount(uint64_t count) {
+        std::unique_lock<std::mutex> lock(global_);
+        total_count_ += count;
```
Comment (jiaweizone) [inline src/data_cell/graph_interface.h:80]: Can we use atomic increment here?

File: src/data_cell/graph_interface.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    virtual void
+    SetTotalCount(uint64_t total_count) {
+        this->total_count_ = total_count;
+    };
+
+    virtual void
+    SetMaxCapacity(uint64_t capacity) {
+        this->max_capacity_ = std::max(capacity, this->total_count_);
```
Comment (jiaweizone) [inline src/data_cell/graph_interface.h:110]: `max_capacity_` equals `total_count_` ?

File: src/data_cell/graph_interface.h (hunk 1 above)
Comment (LHT129) [reply src/data_cell/graph_interface.h:80]: now use lock in external Index, we will apply atomic in the datacell layer or add concurrent interface in future

File: src/data_cell/graph_interface.h (hunk 2 above)
Comment (LHT129) [reply src/data_cell/graph_interface.h:110]: max_capacity_  >=  total_count_, equal is allowed

**Pull request merged**